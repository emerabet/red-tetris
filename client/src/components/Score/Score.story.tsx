import React from 'react';
import { storiesOf } from '@storybook/react';
import Score from './Score';

storiesOf('Score', module)
   .add('999', () => (
      <Score
        score={999}
      />
   ));
