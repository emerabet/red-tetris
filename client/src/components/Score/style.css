.scoreContainer {
    color: #fff;
    font-family: 'Short Stack', cursive;
    font-size: 28px;
}