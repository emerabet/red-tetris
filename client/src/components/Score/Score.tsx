import React, { Component, useState, useEffect } from 'react';

import './style.css';

interface ScoreProps {
  level:number;
  score: number;
  small?: boolean;
}

const Score: React.SFC<ScoreProps> = (props) => {

  return (
    <div className={props.small ? 'scoreContainerSmall' : 'scoreContainer'}>
    <div>LEVEL: {props.level}</div>
      <div>SCORE: {props.score}</div>
    </div>

  );
};

export default Score;
