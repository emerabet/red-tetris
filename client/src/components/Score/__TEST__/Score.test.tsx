import React from 'react';
import renderer from 'react-test-renderer';
import Score from '../Score';

it('renders correctly with defaults', () => {
  const score = renderer.create(<Score
    level={4}
    score={999}
    count={2}
    small
    username="test"
    action="joined"
  />).toJSON();
  expect(score).toMatchSnapshot();
});

it('renders correctly with defaults', () => {
  const score = renderer.create(<Score
    level={4}
    score={999}
    count={2}
    username="test"
    action="joined"
  />).toJSON();
  expect(score).toMatchSnapshot();
});
