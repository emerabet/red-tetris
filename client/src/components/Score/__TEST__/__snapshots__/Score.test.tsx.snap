// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<div
  className="scoreContainer"
>
  <div>
    LEVEL: 
    4
  </div>
  <div>
    SCORE: 
    999
  </div>
</div>
`;
